tools/src/launcher/Main.hs

```diff
@@ -515,11 +519,47 @@ spawnNode (path, args, mbLogPath) = do
             logInfo "Node has started"
             return (ph, asc)
 
-runWallet :: (FilePath, [Text]) -> M ExitCode
-runWallet (path, args) = do
+runWallet :: Bool -> (FilePath, [Text], Maybe FilePath) -> M ExitCode
+runWallet shouldLog (path, args, mLogPath) = do
     logNotice "Starting the wallet"
-    view _1 <$>
-        liftIO (readProcessWithExitCode path (map toString args) mempty)
+    phvar <- newEmptyMVar
+    liftIO $ case mLogPath of
+        Just lp -> do
+            createDirectoryIfMissing True (directory lp)
+            (_, logHandle) <- (lp,) <$> openFile lp AppendMode
+            liftIO $ IO.hSetBuffering logHandle IO.LineBuffering
+            let cr = createProc (Process.UseHandle logHandle) path args
+            system' phvar cr mempty EWallet
+        Nothing ->
+           let cr = if shouldLog then
+                        Process.Inherit
+                    else Process.NoStream
+           in liftIO $ system' phvar (createProc cr path args) mempty EWallet
+
+createProc :: Process.StdStream -> FilePath -> [Text] -> Process.CreateProcess
+createProc stdStream path args =
+    (Process.proc path (map toString args))
+            { Process.std_in = Process.CreatePipe
+            , Process.std_out = stdStream
+            , Process.std_err = stdStream
+            }
+
+asyncLog :: Handle -> Handle -> ProcessHandle -> Executable -> IO ExitCode
+asyncLog stdO stdE pH nt = do
+    withAsync (forever $ IO.hGetLine stdO >>= customLogger LInfo nt) $ \_ ->
+        withAsync (forever $ IO.hGetLine stdE >>= customLogger LError nt) $ \_ -> do
+        waitForProcess pH
+
+customLogger :: Log.CanLog m => LType -> Executable -> String -> m ()
+customLogger logType logName logStr = do
+    Log.usingLoggerName (Log.LoggerName logNameStr) $
+        case logType of
+            LInfo  -> logInfo   $ toText logStr
+            LError -> logError  $ toText logStr
+    where
+        logNameStr = case logName of
+            ENode   -> "node"
+            EWallet -> "wallet"
```

Comment:
Btw, I didn't manage to make this `customLogger` work. Maybe I did something wrong, I will re-test after you fix everything I pointed so far.
I suspect that's because `node` and `wallet` logger names have default severity, which is `Warning`.
And to be honest, I don't see why it's necessary to use `log-warper`'s logging here, why not just use simple `putStrLn`? You can add desired prefix using `mappend`.

Reply:
I used `log-warper` since I noticed it was introduced to this file some time ago. If the logs need to be colored I will revert back.
https://imgur.com/a/kihiN